### PR TITLE
Added context output and indicator tagging to CreateIndicatorsFromStix

### PR DESCRIPTION
**Pull request by mohlcyber**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Added argument *context* argument for context output to ***!CreateIndicatorsFromSTIX***
- Added argument *tags* argument for indicator tagging to ***!CreateIndicatorsFromSTIX***

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 1]
```diff
@@ -2,19 +2,31 @@ args:
...
   required: true
+- auto: PREDEFINED
+  defaultValue: "False"
+  description: Adds Indicators to context.
+  name: context
+  predefined:
+  - "True"
+  - "False"
+- description: Adds tags to Indicators. Comma separated.
+  name: tags
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:13]:
```suggestion
- auto: PREDEFINED
  defaultValue: "false"
  description: Adds Indicators to context.
  name: add_context
  predefined:
  - "true"
  - "false"
- description: Adds tags to Indicators. Comma separated.
  name: tags
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py [hunk 2]
```diff
@@ -34,34 +34,57 @@ def create_relationships_from_entity(indicator_relationships):
...
+        context_data.append(context)
+
         if is_error(res[0]):
             errors.append(f'Error creating indicator - {(res[0]["Contents"])}')
-    result = CommandResults(
-        readable_output=f"Create Indicators From STIX: {len(indicators) - len(errors)} indicators were created.",
-        relationships=relationships_objects
-    )
+
+    if args.get('context') == 'True':
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:67]:
```suggestion
    if add_context:
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py [hunk 3]
```diff
@@ -34,34 +34,57 @@ def create_relationships_from_entity(indicator_relationships):
...
-def create_indicators_loop(indicators):
+def create_indicators_loop(args, indicators):
     """ Create indicators using createNewIndicator automation
 
     :param indicators: parsed indicators
     :return: errors if exist
     """
     relationships_objects = list()
     errors = list()
+    context_data = []
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:45]:
```suggestion
    context_data = []
    add_context = argToBoolean(args.get('add_context', False))
    tags = argToList(args.get('tags'))
```
please use `argToBoolean` and `argToList` methods

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 4]
```diff
@@ -2,19 +2,31 @@ args:
...
+runas: DBotWeakRole
+runonce: false
 script: '-'
+scripttarget: 0
 subtype: python3
 tags:
 - stix
 - ioc
-timeout: '0'
 type: python
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:32]: please remove all the changes from the yml file except:
`dockerimage: demisto/python3:3.10.12.63474`

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py [hunk 5]
```diff
@@ -34,34 +34,57 @@ def create_relationships_from_entity(indicator_relationships):
...
-        relationships=relationships_objects
-    )
+
+    if args.get('context') == 'True':
+        result = CommandResults(
+            readable_output=f"Create Indicators From STIX: {len(indicators) - len(errors)} indicators were created.",
+            relationships=relationships_objects,
+            outputs_prefix='StixIndicators',
+            outputs=context_data
+        )
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:73]:
```suggestion
        result = CommandResults(
            readable_output=f"Create Indicators From STIX: {len(indicators) - len(errors)} indicators were created.",
            relationships=relationships_objects,
            outputs_prefix='StixIndicators',
            outputs_key_field='value'
            outputs=context_data
        )
```

File: Packs/CommonScripts/ReleaseNotes/1_12_50.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### CreateIndicatorsFromSTIX
+
+- Added argument *context* argument for context output to ***!CreateIndicatorsFromSTIX***
+- Added argument *tags* argument for indicator tagging to ***!CreateIndicatorsFromSTIX***
```
Comment (adi88d) [inline Packs/CommonScripts/ReleaseNotes/1_12_50.md:7]:
```suggestion
- Added  *add_context* and * tags* arguments for context output.
- Updated the Docker image to: *demisto/python3:3.10.12.63474*.
```

**Review by adi88d — CHANGES_REQUESTED** [review]
Great Job!
Thanks a lot for this contribution, It is highly appreciated.
I added some review comments and suggestion:
- Please fix the failed tests in `CreateIndicatorsFromSTIX_test.py` file.
- Add to the file also tests for `create_indicators_loop` method to check the results with/ without context and tags arguments.
- see my other comments below

**Comment by mohlcyber** [comment]
Alright seems to be ok. Please let me know if you need any additional information / changes from my side. Thanks so much @adi88d .

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py [hunk 7]
```diff
@@ -34,34 +34,59 @@ def create_relationships_from_entity(indicator_relationships):
...
             relationships_objects.extend(relationship_object)
+
+        if len(tags) > 0:
+            if 'customFields' in indicator:
+                if 'tags' in indicator.get('customFields'):
+                    [indicator['customFields']['tags'].append(ntag) for ntag in tags]
+
         res = demisto.executeCommand("createNewIndicator", indicator)
+
+        context = {'value': indicator['value'], 'type': indicator['type']}
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:62]: consider adding the tags to the context too

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 8]
```diff
@@ -2,6 +2,15 @@ args:
...
   required: true
+- auto: PREDEFINED
+  defaultValue: "false"
+  description: Adds Indicators to context.
+  name: add_context
+  predefined:
+  - "true"
+  - "false"
+- description: Adds tags to Indicators. Comma separated.
+  name: tags
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:13]: please add the `StixIndicators.type` and `StixIndicators.value` to the outputs

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX_test.py [hunk 9]
```diff
@@ -15,10 +12,35 @@ def test_create_indicators_loop(mocker):
...
+    args = {
+        'add_context': 'true',
+        'tags': 'tag1,tag2'
+    }
+
+    with open('test_data/expected_result_example3.json') as json_f:
+        indicators = json.load(json_f)
+    mocker.patch.object(demisto, 'executeCommand', return_value=[None])
+    results, errors = create_indicators_loop(args, indicators)
     assert errors == []
```
Comment (adi88d) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX_test.py:44]: please add a test for `create_indicators_loop` with tags and `add_context=true` and check the context created and the tags added

**Review by adi88d — CHANGES_REQUESTED** [review]
Great! please update the `CreateIndicatorsFromSTIX/README.md` file with the new arguments and the outpus

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py (hunk 7 above)
Comment (mohlcyber) [reply Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:62]: Done. Added tags to context output.

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml (hunk 8 above)
Comment (mohlcyber) [reply Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:13]: done as per commit.

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX_test.py (hunk 9 above)
Comment (mohlcyber) [reply Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX_test.py:44]: Added as part of commit.

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 10]
```diff
@@ -2,6 +2,22 @@ args:
 - description: The entry ID of the STIX file.
   name: entry_id
   required: true
+- auto: PREDEFINED
+  defaultValue: "false"
+  description: Adds Indicators to context.
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:7]:
```suggestion
  description: Adds indicators to context.
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 11]
```diff
@@ -2,6 +2,22 @@ args:
...
+  name: add_context
+  predefined:
+  - "true"
+  - "false"
+- description: Adds tags to Indicators. Comma separated.
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:12]:
```suggestion
- description: A comma-separated list of tags to add to indicators. 
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml [hunk 12]
```diff
@@ -2,6 +2,22 @@ args:
...
+  name: tags
+outputs:
+- contextPath: StixIndicators.type
+  description: Type of the Indicator.
+- contextPath: StixIndicators.value
+  description: Value of the Indicator.
+- contextPath: StixIndicators.tags
+  description: Tags of the Indicator.
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml:20]:
```suggestion
  description: Type of the indicator.
- contextPath: StixIndicators.value
  description: Value of the indicator.
- contextPath: StixIndicators.tags
  description: Tags of the indicator.
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/README.md [hunk 13]
```diff
@@ -17,7 +17,15 @@ Wrapper for the **StixParser** automation. This automation creates indicators an
 | **Argument Name** | **Description** |
 | --- | --- |
 | entry_id | The entry ID of the STIX file. |
+| add_context | Adds Indicators to context. |
+| tags | Adds tags to Indicators. Comma separated. |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/README.md:21]:
```suggestion
| add_context | Adds indicators to context. |
| tags | A comma-separated list of tags to add to indicators.  |
```

File: Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/README.md [hunk 14]
```diff
@@ -17,7 +17,15 @@ Wrapper for the **StixParser** automation. This automation creates indicators an
...
 ## Outputs
+
 ---
-There are no outputs for this script.
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| StixIndicators.type | Type of the Indicator | String |
+| StixIndicators.value | Value of the Indicator | String |
+| StixIndicators.tags | Tags of the Indicator | Unknown |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/README.md:31]:
```suggestion
| StixIndicators.type | Type of the indicator. | String |
| StixIndicators.value | Value of the indicator. | String |
| StixIndicators.tags | Tags of the indicator. | Unknown |
```

**Review by adi88d — APPROVED** [review]
Nice job!

**Pull request merged**